### PR TITLE
<DataTable/> - updated sorting behaviour

### PR DESCRIPTION
**Pull request by yardnsm**
Continued from https://github.com/wix/wix-style-react/pull/2433.

### What changed

This PR continues from the changes made in https://github.com/wix/wix-style-react/pull/2433.

* Removed the the `sortIconDirectionOnHover` prop. From now on, the `<DataTable/>` will determine the next direction itself.
* The next suggested sorting direction will be passed in the `onSortClick ` callback.
* Streamlined the sorting example in the `<Table/>` story.

### Todos

- [ ] Add e2e tests (mainly for the hover behaviour)
- [ ] Document Document Document!

### CODE REVIEW

File: src/DataTable/DataTable.driver.js [hunk 1]
```diff
@@ -71,7 +75,23 @@ const dataTableDriverFactory = ({element, wrapper, component}) => {
...
     clickSort: (index, eventData) => ReactTestUtils.Simulate.click(getHeaderCell(index), eventData),
-    getRowDetails: index => getRowDetails(index)
+    getRowDetails: index => getRowDetails(index),
+
+    // New sorting arrows
+    getActiveSortingArrowDirection: index => {
+      const hooks = getSortingIconsDataHooks(index);
+      const activeIconDataHook = hooks.find(h => h.includes('active'));
+
+      return activeIconDataHook && /active_arrow_(.+)$/.exec(activeIconDataHook)[1];
```
Comment (yardnsm) [inline src/DataTable/DataTable.driver.js]: @hepiyellow @shlomitc Do you think its a trivial way to retrieve this kind of data?

File: src/DataTable/DataTable.js [hunk 2]
```diff
@@ -344,10 +429,34 @@ class TableHeader extends Component {
       [this.style.thText]: this.props.newDesign
     });
 
-    const optionalHeaderCellProps = {};
+    let optionalHeaderCellProps = {};
+
+    let sortIconDirection = column.sortIconDirection;
+    let useLegacySortingBehaviour = false;
```
Comment (yardnsm) [inline src/DataTable/DataTable.js]: I hate this `useLegacySortingBehaviour`. Do you guys have anther suggestion for this?

File: src/DataTable/DataTable.spec.js [hunk 3]
```diff
@@ -388,17 +388,175 @@ describe('Table', () => {
...
+       * suggested one. This would cause an issue when the user still hovers
+       * the row, as the last direction that was shown on hover is gone.
+       *
+       * A solution for this is to store the previous icon direction when the
+       * user *clicks* the row, and mark the clicked row as "active" untill the
+       * user exists it. As long as the row is "active", the previous icon
+       * direction is preserved.
+       */
+
+      class ControlledSortingExample extends React.Component {
```
Comment (yardnsm) [inline src/DataTable/DataTable.spec.js]: Is it the right place for this component?

File: src/DataTable/DataTable.js [hunk 4]
```diff
@@ -344,10 +429,34 @@ class TableHeader extends Component {
...
+    if (sortIconDirection === undefined) {
+      sortIconDirection = (
+        column.sortDescending === undefined ? undefined : (
+          column.sortDescending ? 'desc' : 'asc'
+        )
+      );
+
+      useLegacySortingBehaviour = true;
+
+      deprecationLog(
```
Comment (hepiyellow) [inline src/DataTable/DataTable.js]: This `deprecationLog ` can be as PropTypes custom validator. so it wouldn't fire every render.

File: src/DataTable/DataTable.js (hunk 2 above)
Comment (hepiyellow) [reply src/DataTable/DataTable.js]: Well you can move this line and the next `if (sortIconDirection === undefined) {`
into the `renderSortingArrow` function.

**Comment by hepiyellow** [comment]
@yardnsm Is this WIP? of Paused?
If Pause-Until-Further-Notice, then please close it (and keep branch)

**Comment by yardnsm** [comment]
Closed due to inactivity. We may consider re-open this for the next major release.

**Pull request closed without merging**